src/urdf/UrdfJoint.ts

```
@@ -47,5 +51,18 @@ export default class UrdfJoint {
     if (origins.length > 0) {
       this.origin = parseUrdfOrigin(origins[0]);
     }
+
+    const axis = xml.getElementsByTagName(UrdfAttrs.Axis);
+    if (axis.length > 0) {
+      const xyzValue = axis[0].getAttribute(UrdfAttrs.Xyz)?.split(' ');
+      if (xyzValue && xyzValue.length === 3) {
```

The `axis` element itself is optional, however according to the URDF spec the `xyz` attribute is mandatory if `axis` exists.

I think we should fail hard (throw) if xyzValue is null or malformed.

---

okay, make sense!